### ISSUE TITLE
phx-disable-with fires on phx-validate event

### ISSUE THREAD

**Issue by benwilson512** [phoenixframework/phoenix_live_view#770]
### Environment

* Elixir version (elixir -v):
```
Erlang/OTP 22 [erts-10.6.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]

Elixir 1.10.0 (compiled with Erlang/OTP 22)
```
* Phoenix version (mix deps): 1.4.16
* Phoenix LiveView version (mix deps): 0.11.0
* NodeJS version (node -v): v10.4.1
* NPM version (npm -v): 6.10.3
* Operating system: OS X
* Browsers you attempted to reproduce this bug on (the more the merrier): Chrome
* Does the problem persist after removing "assets/node_modules" and trying again? Yes/no: Yes

### Actual Behavior
https://github.com/benwilson512/bug_example/commit/2cebdc52170407e3ab870ae8421b7a70501472a5

This is easiest to observe if you add some latency via `liveSocket.enableLatencySim(1000)`. 

Given this form:
```
    <form phx-change="update" phx-disable-with="Saving...">
    <input type="text" name="text" value="<%= @text %>">
    </form>
```

### Expected Behaviour

`phx-disable-with` should only fire on submit events.

**Comment by chrismccord** [comment]
This is actually a documentation bug as `disable-with` applies to any phx event that is put on the wire. For your usecase, you can instead take advantage of the loading state classes that landed a a couple versions ago:

```
.while-submitting { display: none; }
.inputs { display: block; }

.phx-submit-loading {
  .while-submitting { display: block; }
  .inputs { display: none; }
}

<form phx-change="update">
  <div class="while-submitting">Please waiting while we save our content...</div>
  <div class="inputs">
    <input type="text" name="text" value="<%= @text %>">
  </div>
</form>
```

I'll bump the docs. Thanks!

**Comment by benwilson512** [comment]
Hey Chris thanks for the update. Is there a way to get the previous functionality this offered? The primary benefit of `phx-disable-with` in the past is that, as I understood it at least, it disabled the actual form button so that you couldn't accidentally double submit by rapidly clicking or hitting enter. The text provided to `phx-disable-with` even updated the actual submit button text. CSS changes aren't really enough to accomplish that.

I can create a dedicated `phx-hook` for this, but in terms of avoiding foot guns, it was nice to have a built in "avoid double submit" thing.

Relatedly, can you elaborate on when `phx-disable-with` is usable with `phx-update`? Particularly given that it bumps focus, it doesn't seem like the two can actually be used together.

**Comment by chrismccord** [comment]
disable-with on the submit button indeed gives you the behaviour you want (disables the button during form submit):

    <button type="submit" phx-disable-with="Saving...">Save</button>

So the css bit is for swapping the form contents, not disabling the button. Make sense?

**Comment by benwilson512** [comment]
Ah! So I should just relocate it, that was the bit I was missing. Thanks!

**Comment by chrismccord** [comment]
Yes, and to expand, any time we submit a form, regardless of phx-disable-with, we disable all buttons and mark all inputs as read only. So you shouldn't be able to submit multiple times while waiting on an acknowledgement regardless of disable-with usage. Let me know if you are seeing otherwise.

**Comment by blackham** [comment]
I was having issues with the multiple submit events being processed.  I worked around it by adding "pointer-events: none" to .phx-click-loading. I should point out I also have materialize-css loaded. And that I was binding phx-click to an anchor <a> tag. I'll try again with <button> later. 

```
.phx-click-loading {  
  opacity: 0.5;       
  transition: opacity 1s ease-out;
  pointer-events: none;                                                                                                                    
} 
```
